helix-term/tests/test/commands.rs

```diff
@@ -133,3 +135,65 @@ async fn test_selection_duplication() -> anyhow::Result<()> {
     .await?;
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_goto_file_impl() -> anyhow::Result<()> {
+    let file = tempfile::NamedTempFile::new()?;
+
+    fn match_paths(app: &Application, matches: Vec<&str>) -> usize {
+        app.editor
+            .documents()
+            .map(|d| d.path())
+            .flatten()
+            .map(|p| p.file_name())
+            .flatten()
+            .filter(|n| matches.iter().any(|m| &OsStr::new(m) == n))
+            .count()
```

Comment:
Clippy has some complaints about this block:

```suggestion
        app.editor
            .documents()
            .filter_map(|d| d.path()?.file_name())
            .filter(|n| matches.iter().any(|m| *m == n.to_string_lossy()))
            .count()
```

then you can also remove the import of OsStr